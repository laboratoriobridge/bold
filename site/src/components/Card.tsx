import { Cell, HFlow, Icon, IconImage, Text, Theme, useStyles } from 'bold-ui'
import React from 'react'
export interface CardProps {
  icon: IconImage
  children: React.ReactNode
}

export function Card(props: CardProps) {
  const { icon, children } = props
  const { classes } = useStyles(createStyles)

  return (
    <div className={classes.card}>
      <HFlow alignItems='center' hSpacing={0.25}>
        <Cell>
          <Icon icon={icon} fill='primary' size={3} />
        </Cell>
        <Cell>
          <Text component='p' fontSize={1.15} color='primary'>
            {children}
          </Text>
        </Cell>
      </HFlow>
    </div>
  )
}

const createStyles = (theme: Theme) => ({
  card: {
    color: theme.pallete.gray.c10,
    backgroundColor: theme.pallete.primary.c90,
    borderRadius: 4,
    maxWidth: '850px',
    margin: '1rem 0',
    padding: '0.75rem',
    border: `1px solid ${theme.pallete.primary.c40}`,
  } as React.CSSProperties,
})
