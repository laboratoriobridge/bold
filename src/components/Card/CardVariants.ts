import { ExternalStyles, focusBoxShadow, hexToRGB, Theme } from '../../styles'

export type CardVariant = 'outline' | 'float' | 'plain'

export const createBaseStyles = (theme: Theme) => ({
  card: {
    width: '100%',
    border: 0,
    borderRadius: `${theme.radius.popper}px`,
    padding: '1rem',
    transition: 'all .2s',
  } as ExternalStyles,
  cardDisabled: {
    cursor: 'not-allowed',
    '& *': {
      pointerEvents: 'none',
    },
  } as ExternalStyles,
  innerDisabled: {
    opacity: 0.4,
  } as ExternalStyles,
})

export const createVariantStyles = (theme: Theme): { [key in CardVariant]: ExternalStyles } => ({
  float: {
    background: theme.pallete.surface.main,
    border: `1px solid ${theme.pallete.gray.c80}`,
    boxShadow: theme.shadows.outer[40],
    '&[data-invalid="true"]': {
      borderColor: theme.pallete.status.danger.main,
    },
    '&[data-disabled="true"]': {
      borderColor: theme.pallete.gray.c80,
      background: theme.pallete.surface.main,
    },
  } as ExternalStyles,
  outline: {
    background: theme.pallete.surface.main,
    border: `1px solid ${theme.pallete.gray.c80}`,
    borderRadius: `${theme.radius.popper}px`,
    '&[data-invalid="true"]': {
      borderColor: theme.pallete.status.danger.main,
    },
    '&[data-disabled="true"]': {
      borderColor: theme.pallete.gray.c80,
      background: theme.pallete.surface.main,
    },
  } as ExternalStyles,
  plain: {
    background: theme.pallete.surface.main,
    borderRadius: `${theme.radius.popper}px`,
    '&[data-invalid="true"]': {
      background: theme.pallete.status.danger.c90,
    },
    '&[data-disabled="true"]': {
      borderColor: theme.pallete.gray.c80,
      background: theme.pallete.surface.main,
    },
  } as ExternalStyles,
})

export const createClickableStyles = (theme: Theme): ExternalStyles => ({
  ...theme.typography.variant('main'),
  cursor: 'pointer',
  textAlign: 'initial',
  '&[data-variant="outline"], &[data-variant="plain"]': {
    borderColor: theme.pallete.gray.c60,
  },
  ':not(:disabled):hover': {
    background: hexToRGB(theme.pallete.gray.c50, 0.1),
    '&[data-variant="float"]': {
      boxShadow: theme.shadows.outer[160],
    },
  },
  ':focus': {
    outline: 'none',
    boxShadow: focusBoxShadow(theme),
    '&[data-variant="float"]': {
      boxShadow: `${theme.shadows.outer[40]}, ${focusBoxShadow(theme)}`,
    },
  },
  ':not(:disabled):not(:active):hover:focus': {
    '&[data-variant="float"]': {
      boxShadow: `${theme.shadows.outer[160]}, ${focusBoxShadow(theme)}`,
    },
  },
  ':not(:disabled):active': {
    boxShadow: theme.shadows.inner['10'],
    '&[data-variant="float"]': {
      // use zero box shadow to keep transition smooth
      boxShadow: `0 0 0 0 rgba(0, 0, 0, 0), 0 0 0 0 rgba(0, 0, 0, 0), 0 0 0 0 rgba(0, 0, 0, 0), ${theme.shadows.inner['10']}`,
    },
  },
  '&[data-selected="true"]': {
    background: theme.pallete.primary.c90,
    borderColor: theme.pallete.primary.main,
    ':not(:disabled):hover': {
      background: theme.pallete.primary.c80,
    },
    '&[data-invalid="true"]': {
      background: theme.pallete.status.danger.c90,
    },
  },
  '&[data-invalid="true"]': {
    borderColor: theme.pallete.status.danger.main,
    ':not(:disabled):hover': {
      background: theme.pallete.status.danger.c80,
    },
    ':focus': {
      outline: 'none',
      boxShadow: focusBoxShadow(theme, 'danger'),
    },
    ':not(:disabled):active': {
      boxShadow: theme.shadows.inner['10'],
    },
    ':not(:disabled):not(:active):hover:focus': {
      '&[data-variant="float"]': {
        boxShadow: `${theme.shadows.outer[160]}, ${focusBoxShadow(theme, 'danger')}`,
      },
    },
  },
  ':disabled': {
    borderColor: theme.pallete.gray.c80,
    '&[data-selected="true"]': {
      borderColor: theme.pallete.primary.c80,
    },
    '&[data-invalid="true"]': {
      borderColor: theme.pallete.gray.c80,
      background: theme.pallete.surface.main,
      '&[data-selected="true"]': {
        borderColor: theme.pallete.status.danger.c80,
        background: theme.pallete.status.danger.c90,
      },
    },
  },
})
