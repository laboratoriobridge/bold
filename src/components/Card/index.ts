export { Card, CardProps } from './Card'
export { ClickableCard, ClickableCardProps } from './ClickableCard'
export { CardVariant } from './cardVariants'
