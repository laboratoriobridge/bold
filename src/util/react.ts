import { MutableRefObject, Ref } from 'react'

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}
