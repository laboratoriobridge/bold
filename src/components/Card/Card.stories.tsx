import React from 'react'
import { action } from '@storybook/addon-actions'
import { boolean, select } from '@storybook/addon-knobs'
import { VFlow } from '../VFlow'
import { HeadingSection } from '../HeadingSection'
import { HFlow } from '../HFlow'
import { Checkbox } from '../Checkbox'
import { Button } from '../Button'
import { Icon } from '../Icon'
import { Text } from '../Text'
import { Tag } from '../Tag'
import { Heading } from '../Heading'
import { Status } from '../Status'
import { Card, CardVariant } from './Card'

const variants: { [key in CardVariant]: CardVariant } = {
  outline: 'outline',
  float: 'float',
  plain: 'plain',
}

const Content = () => (
  <HFlow hSpacing={0.5} style={{ gridAutoColumns: 'auto 1fr' }}>
    <Checkbox />
    <HFlow alignItems='center' justifyContent='space-between'>
      <VFlow vSpacing={0.5}>
        <VFlow vSpacing={0.25}>
          <HFlow hSpacing={0.5} alignItems='center'>
            <Heading level={4}>Title</Heading>
            <Tag>Tag label</Tag>
          </HFlow>
          <Text>Description</Text>
        </VFlow>
        <Status type='info' text='Status description' />
      </VFlow>
      <Button skin='ghost' size='small'>
        <Icon icon='adjust' />
      </Button>
    </HFlow>
  </HFlow>
)

export default {
  title: 'Components/Card',
}

export const Default = () => {
  const variant = select('variant', variants, 'outline')
  const selected = boolean('selected', false)
  const invalid = boolean('invalid', false)
  const disabled = boolean('disabled', false)

  return (
    <VFlow>
      <HeadingSection level={2} vSpace={8} title='Interactive'>
        <Card
          variant={variant}
          selected={selected}
          disabled={disabled}
          error={invalid}
          onClick={action('Card clicked')}
        >
          <Content />
        </Card>
      </HeadingSection>
      <HeadingSection level={2} vSpace={8} title='Non-interactive'>
        <Card variant={variant} selected={selected} disabled={disabled} error={invalid}>
          <Content />
        </Card>
      </HeadingSection>
    </VFlow>
  )
}

export const Variants = () => {
  const invalid = boolean('invalid', false)

  return (
    <VFlow>
      <HeadingSection level={2} vSpace={8} title='Outline card'>
        <Card variant='outline' error={invalid}>
          <Content />
        </Card>
      </HeadingSection>
      <HeadingSection level={2} vSpace={8} title='Float card'>
        <Card variant='float' error={invalid}>
          <Content />
        </Card>
      </HeadingSection>
      <HeadingSection level={2} vSpace={8} title='Plain card'>
        <Card variant='plain' error={invalid}>
          <Content />
        </Card>
      </HeadingSection>
    </VFlow>
  )
}

export const Clickable = () => {
  const selected = boolean('selected', false)
  const invalid = boolean('invalid', false)
  const disabled = boolean('disabled', false)

  return (
    <VFlow>
      <HeadingSection level={2} vSpace={8} title='Outline card'>
        <Card
          variant='outline'
          selected={selected}
          disabled={disabled}
          error={invalid}
          onClick={action('Card clicked')}
        >
          <Content />
        </Card>
      </HeadingSection>
      <HeadingSection level={2} vSpace={8} title='Float card'>
        <Card variant='float' selected={selected} disabled={disabled} error={invalid} onClick={action('Card clicked')}>
          <Content />
        </Card>
      </HeadingSection>
      <HeadingSection level={2} vSpace={8} title='Plain card'>
        <Card variant='plain' selected={selected} disabled={disabled} error={invalid} onClick={action('Card clicked')}>
          <Content />
        </Card>
      </HeadingSection>
    </VFlow>
  )
}
