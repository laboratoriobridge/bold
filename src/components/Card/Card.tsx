import React from 'react'

import { css } from 'emotion'
import { ExternalStyles, Theme, useTheme } from '../../styles'
import { CardVariant, createBaseStyles, createVariantStyles } from './CardVariants'

export interface CardProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'style'> {
  style?: ExternalStyles

  /**
   * Defines the visual style of the Card
   * @default 'outline'
   */
  variant?: CardVariant

  /**
   * Specify whether the Card should have error appearance
   */
  invalid?: boolean

  /**
   * Specify whether the Card should have disabled appearance
   */
  disabled?: boolean
}

export function Card(props: CardProps) {
  const { variant = 'outline', children, invalid, disabled, style, ...rest } = props

  const theme = useTheme()
  const styles = createStyles(theme, variant, disabled, invalid, style)

  return (
    <div className={styles.card} {...rest}>
      <div className={styles.inner}>{children}</div>
    </div>
  )
}

const createStyles = (
  theme: Theme,
  variant: CardVariant,
  disabled: boolean,
  invalid: boolean,
  externalStyles: ExternalStyles
) => {
  const baseStyles = createBaseStyles(theme)
  const variantStyles = createVariantStyles(theme)[variant]

  return {
    card: css(
      baseStyles.card,
      variantStyles.base,
      invalid && variantStyles.invalid,
      disabled && baseStyles.cardDisabled,
      disabled && variantStyles.disabled,
      externalStyles
    ),
    inner: css(disabled && baseStyles.innerDisabled),
  }
}
