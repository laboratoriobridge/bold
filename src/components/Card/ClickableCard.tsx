import React, { ComponentProps } from 'react'
import { ExternalStyles, useCss } from '../../styles'
import { CardVariant, createBaseStyles, createClickableStyles, createVariantStyles } from './CardVariants'

export interface ClickableCardProps extends Omit<ComponentProps<'button'>, 'style'> {
  style?: ExternalStyles

  /**
   * Defines the visual style of the Card
   * @default 'outline'
   */
  variant?: CardVariant

  /**
   * Specify whether the Card should be selected
   */
  selected?: boolean

  /**
   * Specify whether the Card should have a error appearance
   */
  invalid?: boolean

  /**
   * Specify whether the Card should be disabled
   */
  disabled?: boolean
}

export function ClickableCard(props: ClickableCardProps) {
  const { variant = 'outline', children, invalid, disabled, selected, style, ...buttonProps } = props

  const { theme, css } = useCss()

  const baseStyles = createBaseStyles(theme)
  const variantStyles = createVariantStyles(theme)[variant]
  const clickableStyles = createClickableStyles(theme)

  const classes = css(
    baseStyles.card,
    variantStyles.base,
    clickableStyles,
    invalid && variantStyles.invalid,
    disabled && baseStyles.cardDisabled,
    disabled && variantStyles.disabled,
    style
  )

  return (
    <button
      type='button'
      className={classes}
      disabled={disabled}
      data-variant={variant}
      data-selected={selected}
      data-invalid={invalid}
      aria-pressed={selected}
      {...buttonProps}
    >
      <div className={css(disabled && baseStyles.innerDisabled)}>{children}</div>
    </button>
  )
}
