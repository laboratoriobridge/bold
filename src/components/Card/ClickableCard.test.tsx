import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { ClickableCard, CardVariant } from '../Card'

it('should render correctly', () => {
  const { container } = render(<ClickableCard>Content</ClickableCard>)
  expect(container).toMatchSnapshot()
})

it('should accept "style" prop', () => {
  const { container } = render(<ClickableCard style={{ background: 'red' }}>Content</ClickableCard>)
  expect(container).toMatchSnapshot()
})

describe('variants', () => {
  it('should render outline variant by default', () => {
    const { getByTestId } = render(<ClickableCard data-testid='clickable-card-outline'>Content</ClickableCard>)
    const card = getByTestId('clickable-card-outline')
    expect(getComputedStyle(card).background).toBe('rgb(255, 255, 255)')
    expect(getComputedStyle(card).borderColor).toBe('#8f8fa2')
    expect(getComputedStyle(card).boxShadow).toBe('')
  })

  it('should render float variant styles when specified', () => {
    const { getByTestId } = render(
      <ClickableCard variant='float' data-testid='clickable-card-float'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-float')
    expect(getComputedStyle(card).background).toBe('rgb(255, 255, 255)')
    expect(getComputedStyle(card).border).toBe('1px solid #d3d4dd')
    expect(getComputedStyle(card).boxShadow).toBe(
      '0 2px 4px -1px rgba(0,0,0,0.2),0 1px 10px 0 rgba(0,0,0,0.12),0 4px 5px 0 rgba(0,0,0,0.14)'
    )
  })

  it('should render plain variant styles when specified', () => {
    const { getByTestId } = render(
      <ClickableCard variant='plain' data-testid='clickable-card-plain'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-plain')
    expect(getComputedStyle(card).background).toBe('rgb(255, 255, 255)')
    expect(getComputedStyle(card).border).toBe('0px')
    expect(getComputedStyle(card).boxShadow).toBe('')
  })
})

describe('click events', () => {
  it('should trigger click event when clicked', () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(
      <ClickableCard onClick={handleClick} data-testid='clickable-card-click'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-click')
    fireEvent.click(card)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('should not trigger click event when disabled', () => {
    const handleClick = jest.fn()
    const { getByTestId } = render(
      <ClickableCard disabled onClick={handleClick} data-testid='clickable-card-disabled-click'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-disabled-click')
    fireEvent.click(card)
    expect(handleClick).not.toHaveBeenCalled()
  })
})

describe('states', () => {
  it.each<CardVariant>(['outline', 'float'])(
    'should apply selected styles in outline and float variants when is selected',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} selected data-testid='clickable-card-selected'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-selected')
      expect(getComputedStyle(card).borderColor).toBe('#0069d0')
      expect(getComputedStyle(card).background).toBe('rgb(236, 240, 255)')
    }
  )

  it('should apply selected styles in plain variant when is selected', () => {
    const { getByTestId } = render(
      <ClickableCard variant='plain' selected data-testid='clickable-card-selected'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-selected')
    expect(getComputedStyle(card).background).toBe('rgb(236, 240, 255)')
  })

  it.each<CardVariant>(['outline', 'float'])(
    'should apply invalid styles in outline and float variants when error specified',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} error='Error' data-testid='clickable-card-invalid'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-invalid')
      expect(getComputedStyle(card).borderColor).toBe('#d01e29')
      expect(getComputedStyle(card).background).toBe('rgb(255, 255, 255)')
    }
  )

  it('should apply invalid styles in plain variant when error specified', () => {
    const { getByTestId } = render(
      <ClickableCard variant='plain' error='Error' data-testid='clickable-card-invalid'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-invalid')
    expect(getComputedStyle(card).background).toBe('rgb(254, 236, 237)')
  })

  it.each<CardVariant>(['outline', 'float', 'plain'])(
    'should apply disabled styles in each variant when is disabled',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} disabled data-testid='clickable-card-disabled'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-disabled')
      expect(getComputedStyle(card).cursor).toBe('not-allowed')
      expect(card).toHaveStyle('border-color: #d3d4dd')
      expect(card).toHaveStyle('background: rgb(255, 255, 255)')
    }
  )

  it.each<CardVariant>(['outline', 'float', 'plain'])(
    'should prioritize disabled styles when the card is disabled and error specified',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} disabled error='Error' data-testid='clickable-card-disabled-invalid'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-disabled-invalid')
      expect(getComputedStyle(card).cursor).toBe('not-allowed')
      expect(card).toHaveStyle('border-color: #d3d4dd')
      expect(card).toHaveStyle('background: rgb(255, 255, 255)')
    }
  )

  it.each<CardVariant>(['outline', 'float'])(
    'should apply correctly styles in outline and float variants when is selected and error specified',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} selected error='Error' data-testid='clickable-card-selected-invalid'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-selected-invalid')
      expect(card).toHaveStyle('border-color: #d01e29')
      expect(card).toHaveStyle('background: rgb(254, 236, 237)')
    }
  )

  it('should apply correctly styles in plain variant when is selected and error specified', () => {
    const { getByTestId } = render(
      <ClickableCard variant='plain' selected error='Error' data-testid='clickable-card-selected-invalid'>
        Content
      </ClickableCard>
    )
    const card = getByTestId('clickable-card-selected-invalid')
    expect(card).toHaveStyle('background: rgb(254, 236, 237)')
  })

  it.each<CardVariant>(['outline', 'float', 'plain'])(
    'should apply correctly styles in each variant when is selected and disabled',
    (variant) => {
      const { getByTestId } = render(
        <ClickableCard variant={variant} selected disabled data-testid='clickable-card-selected-disabled'>
          Content
        </ClickableCard>
      )
      const card = getByTestId('clickable-card-selected-disabled')
      expect(getComputedStyle(card).cursor).toBe('not-allowed')
      expect(card).toHaveStyle('border-color: #c5d4ff')
      expect(card).toHaveStyle('background: rgb(236, 240, 255)')
    }
  )
})
