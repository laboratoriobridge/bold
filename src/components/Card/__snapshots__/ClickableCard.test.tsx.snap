// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept "style" prop 1`] = `
.emotion-1 {
  width: 100%;
  border: 0;
  border-radius: 2px;
  padding: 1rem;
  -webkit-transition: all .2s;
  transition: all .2s;
  background: #FFFFFF;
  border: 1px solid #D3D4DD;
  border-radius: 2px;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
  cursor: pointer;
  text-align: initial;
  background: red;
}

.emotion-1[data-invalid="true"] {
  border-color: #D01E29;
}

.emotion-1[data-disabled="true"] {
  border-color: #D3D4DD;
  background: #FFFFFF;
}

.emotion-1[data-variant="outline"],
.emotion-1[data-variant="plain"] {
  border-color: #8F8FA2;
}

.emotion-1:not(:disabled):hover {
  background: rgba(122,123,142,0.1);
}

.emotion-1:not(:disabled):hover[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:focus[data-variant="float"] {
  box-shadow: 0 2px 4px -1px rgba(0,0,0,0.2),0 1px 10px 0 rgba(0,0,0,0.12),0 4px 5px 0 rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:not(:disabled):not(:active):hover:focus[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1:not(:disabled):active[data-variant="float"] {
  box-shadow: 0 0 0 0 rgba(0,0,0,0),0 0 0 0 rgba(0,0,0,0),0 0 0 0 rgba(0,0,0,0),inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1[data-selected="true"] {
  background: #ECF0FF;
  border-color: #0069D0;
}

.emotion-1[data-selected="true"]:not(:disabled):hover {
  background: #C5D4FF;
}

.emotion-1[data-selected="true"][data-invalid="true"] {
  background: #FEECED;
}

.emotion-1[data-invalid="true"] {
  border-color: #D01E29;
}

.emotion-1[data-invalid="true"]:not(:disabled):hover {
  background: #FAC8C9;
}

.emotion-1[data-invalid="true"]:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #D01E29;
}

.emotion-1[data-invalid="true"]:not(:disabled):active {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1[data-invalid="true"]:not(:disabled):not(:active):hover:focus[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #D01E29;
}

.emotion-1:disabled {
  border-color: #D3D4DD;
}

.emotion-1:disabled[data-selected="true"] {
  border-color: #C5D4FF;
}

.emotion-1:disabled[data-invalid="true"] {
  border-color: #D3D4DD;
  background: #FFFFFF;
}

.emotion-1:disabled[data-invalid="true"][data-selected="true"] {
  border-color: #FAC8C9;
  background: #FEECED;
}

<div>
  <button
    class="emotion-1"
    data-invalid="false"
    data-variant="outline"
    type="button"
  >
    <div
      class="emotion-0"
    >
      Content
    </div>
  </button>
</div>
`;

exports[`should render correctly 1`] = `
.emotion-1 {
  width: 100%;
  border: 0;
  border-radius: 2px;
  padding: 1rem;
  -webkit-transition: all .2s;
  transition: all .2s;
  background: #FFFFFF;
  border: 1px solid #D3D4DD;
  border-radius: 2px;
  font-family: "IBM Plex Sans",sans-serif;
  font-size: 0.8125rem;
  line-height: 1.5;
  color: #24252E;
  cursor: pointer;
  text-align: initial;
}

.emotion-1[data-invalid="true"] {
  border-color: #D01E29;
}

.emotion-1[data-disabled="true"] {
  border-color: #D3D4DD;
  background: #FFFFFF;
}

.emotion-1[data-variant="outline"],
.emotion-1[data-variant="plain"] {
  border-color: #8F8FA2;
}

.emotion-1:not(:disabled):hover {
  background: rgba(122,123,142,0.1);
}

.emotion-1:not(:disabled):hover[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:focus[data-variant="float"] {
  box-shadow: 0 2px 4px -1px rgba(0,0,0,0.2),0 1px 10px 0 rgba(0,0,0,0.12),0 4px 5px 0 rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:not(:disabled):not(:active):hover:focus[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #0069D0;
}

.emotion-1:not(:disabled):active {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1:not(:disabled):active[data-variant="float"] {
  box-shadow: 0 0 0 0 rgba(0,0,0,0),0 0 0 0 rgba(0,0,0,0),0 0 0 0 rgba(0,0,0,0),inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1[data-selected="true"] {
  background: #ECF0FF;
  border-color: #0069D0;
}

.emotion-1[data-selected="true"]:not(:disabled):hover {
  background: #C5D4FF;
}

.emotion-1[data-selected="true"][data-invalid="true"] {
  background: #FEECED;
}

.emotion-1[data-invalid="true"] {
  border-color: #D01E29;
}

.emotion-1[data-invalid="true"]:not(:disabled):hover {
  background: #FAC8C9;
}

.emotion-1[data-invalid="true"]:focus {
  outline: none;
  box-shadow: 0 0 0 2px #F0F0F5,0 0 0 4px #D01E29;
}

.emotion-1[data-invalid="true"]:not(:disabled):active {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.14),inset 0 2px 4px 0 rgba(0,0,0,0.08);
}

.emotion-1[data-invalid="true"]:not(:disabled):not(:active):hover:focus[data-variant="float"] {
  box-shadow: 0 8px 10px -5px rgba(0,0,0,0.2),0 6px 30px 5px rgba(0,0,0,0.12),0 16px 24px 2px rgba(0,0,0,0.14),0 0 0 2px #F0F0F5,0 0 0 4px #D01E29;
}

.emotion-1:disabled {
  border-color: #D3D4DD;
}

.emotion-1:disabled[data-selected="true"] {
  border-color: #C5D4FF;
}

.emotion-1:disabled[data-invalid="true"] {
  border-color: #D3D4DD;
  background: #FFFFFF;
}

.emotion-1:disabled[data-invalid="true"][data-selected="true"] {
  border-color: #FAC8C9;
  background: #FEECED;
}

<div>
  <button
    class="emotion-1"
    data-invalid="false"
    data-variant="outline"
    type="button"
  >
    <div
      class="emotion-0"
    >
      Content
    </div>
  </button>
</div>
`;
